menas/ui/service/MonitoringService.js

```diff
@@ -337,8 +337,8 @@ var MonitoringService = new function() {
       name: original["name"],
       order: original["order"],
       workflowName: original["workflowName"],
-      //startDate: moment(original["processStartTime"], "DD-MM-YYYY HH:mm:ss").toDate(),
-      endDate: moment(original["processEndTime"], "DD-MM-YYYY HH:mm:ss").toDate()
+      //startDate: moment.parseZone(original["processStartTime"], "DD-MM-YYYY HH:mm:ss ZZ").toDate(),
```

Comment:
Where does the `startDate` come from?

Reply:
in the  commented line the 'startDate' is processStartTime of the last checkpoint.
the 'startDateTime' of the entire job is being casted inside mongo query in order to filter and sort the output

Reply:
Isn't that a possible source of discrepancy (as this bug shows), that each is done at different places?
Cannot it be done all at one place?

Reply:
@benedeki:
 - All the date/time fields are currently stored as str in Mongo
- When we query the runs, we have to cast infodate and start time of a job to Mongo's ISODate in order to do filtering, sorting and grouping within the monitoring query. However, it is a costly operation, therefore we only cast the fields that are required.  
- In Web UI, we need to cast date/time fields to DateTime. For those 2 fields we do cast ISODate -> DateTime. For the rest: str - > DateTime. Currently, there is only one later field in use, but more can be added when monitoring evolves.

Reply:
And I guess the original string for start time is not available here anymore to use?
I would consider this a hotfix then. But still a possible source of future problems.

Reply:
@benedeki, in principal it is possible to duplicate those fields, to have them as both as ISODate and original String. However, it may increase discrepancy even more. What do you think?

Reply:
Not sure. For now let's keep it.
Btw, whys is the field in Mongo a string and not a timestamp?

Reply:
> Btw, whys is the field in Mongo a string and not a timestamp?

This is _INFO file transformed into a Run object. JSON does not have a date/time type. So it just seems it was never parsed when moved to Run object